pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsXml.scala

```diff
@@ -1,30 +1,48 @@
-package uk.ac.wellcome.platform.transformer.mets.parsers
+package uk.ac.wellcome.platform.transformer.mets.transformer
 
-import scala.collection.immutable.ListMap
 import scala.util.Try
+import scala.collection.immutable.ListMap
 import scala.xml.{Elem, NodeSeq, XML}
 
-import uk.ac.wellcome.platform.transformer.mets.transformer.Mets
-
-object MetsXmlParser {
-
-  def apply(str: String): Either[Throwable, Mets] =
-    for {
-      is <- Try(XML.loadString(str)).toEither
-      mets <- MetsXmlParser(is)
-    } yield (mets)
+case class MetsXml(root: MetsXmlFile, manifestations: List[MetsXmlFile]) {
 
-  def apply(root: Elem): Either[Exception, Mets] = {
+  def toMetsData: Either[Exception, MetsData] =
     for {
-      id <- recordIdentifier(root)
-      maybeAccessCondition <- accessCondition(root)
+      id <- root.recordIdentifier
+      accessCondition <- root.accessCondition
     } yield
-      Mets(
+      MetsData(
         recordIdentifier = id,
-        accessCondition = maybeAccessCondition,
-        thumbnailLocation = thumbnailLocation(root, id),
+        accessCondition = accessCondition,
+        thumbnailLocation = root.thumbnailLocation(id),
       )
-  }
+
+  def hasManifestations = manifestations.nonEmpty
+}
+
+object MetsXml {
+
+  def apply(root: Elem, manifestations: List[Elem] = Nil): MetsXml =
+    MetsXml(MetsXmlFile(root), manifestations.map(MetsXmlFile(_)))
+
+  def apply(root: String,
+            manifestations: List[String]): Either[Throwable, MetsXml] =
+    MetsXmlFile(root).flatMap { root =>
+      val results = manifestations.map(MetsXmlFile(_))
+      results.partition(_.isLeft) match {
+        case (Nil, manifestations) =>
+          Right(
+            MetsXml(root, manifestations.map(_.right.get))
+          )
+        case (errors, _) =>
+          Left(
+            new Exception("Errors parsing manifestations")
```

Comment:
Is there a way of logging the errors in the message? otherwise, it might become hard debugging parsing errors